tmt/steps/report/reportportal.py

```diff
@@ -49,6 +51,16 @@ class ReportReportPortalData(tmt.steps.report.ReportStepData):
         metavar="NAME",
         default=None,
         help="The launch name (base name of run id used by default).")
+    ca_cert_file: Optional[str] = tmt.utils.field(
+        option="--ca-cert-file",
+        metavar="CA_CERT",
+        default=None,
+        help="Set reportportal cert file")
+    add_meta_data: Optional[str] = tmt.utils.field(
+        option="--add-meta-data",
+        metavar="ADD_METADATA",
+        default=None,
+        help="Set metadata launch { key: key1, value: value1 }")
```

Comment:
What values are allowed to be used as metadata? Can it be anything JSON allows, e.g. `--add-metadata {"foo": [1, false]}`?

If only key/value pairs of some string-ish nature are expected, I would maybe don't even mention `{}` in the help, and focus instead on the `key=value` form of input, i.e. `--add-meta-data foo=bar --add-meta-data baz=79`. With `multiple=True`, it should be easier for the user to enter this kind of values, while still reasonably easy for tmt to put them into a JSON object & upload to RP.

If JSON objects are allowed and expected, then I would recommend adding a sanity check right after line 167, something like this:

```python
try:
  _ = json.loads(self.get('add-meta-data'))
except Exception as exc:
  raise SpecificationError("Invalid syntax of provided metadata") from exc
```

That way, you can detect broken metadata way sooner (`--add-meta-data '{"foo": }'`)

Final note: I'd recommend `metadata` or `meta-data`, i.e. without the "add" prefix. The field can be specified in tmt medata as well as on the command line, and tmt keys tend to take more declarative approach: "metadata" vs "add metadata", "ca cert file" instead of "use ca cert file", "project name" instead of "set project name" and so on. It's perfectly fine to use `--meta-data` on command line, while `add-meta-data: ...` in a plan looks inconsistent.

Reply:
> What values are allowed to be used as metadata? Can it be anything JSON allows, e.g. `--add-metadata {"foo": [1, false]}`?

Yes, as per report portal docs, 
Will have to tests it.
> 
> If only key/value pairs of some string-ish nature are expected, I would maybe don't even mention `{}` in the help, and focus instead on the `key=value` form of input, i.e. `--add-meta-data foo=bar --add-meta-data baz=79`. With `multiple=True`, it should be easier for the user to enter this kind of values, while still reasonably easy for tmt to put them into a JSON object & upload to RP.

ACK

> 
> If JSON objects are allowed and expected, then I would recommend adding a sanity check right after line 167, something like this:
> 
> ```python
> try:
>   _ = json.loads(self.get('add-meta-data'))
> except Exception as exc:
>   raise SpecificationError("Invalid syntax of provided metadata") from exc
> ```
> 
> That way, you can detect broken metadata way sooner (`--add-meta-data '{"foo": }'`)
> 
ACK, thanks 

> Final note: I'd recommend `metadata` or `meta-data`, i.e. without the "add" prefix. The field can be specified in tmt medata as well as on the command line, and tmt keys tend to take more declarative approach: "metadata" vs "add metadata", "ca cert file" instead of "use ca cert file", "project name" instead of "set project name" and so on. It's perfectly fine to use `--meta-data` on command line, while `add-meta-data: ...` in a plan looks inconsistent.

ACK

Reply:
>> What values are allowed to be used as metadata? Can it be anything JSON allows, e.g. --add-metadata {"foo": [1, false]}?
>>
> Yes, as per report portal docs,
> Will have to tests it.

Reply:
Done